docs/user_guide.md

```diff
@@ -1227,16 +1227,36 @@ If you see this error:
 ***WARNING*** CPU scaling is enabled, the benchmark real time measurements may be noisy and will incur extra overhead.
 ```
 
-you might want to disable the CPU frequency scaling while running the benchmark using a script like:
+you might want to disable the CPU frequency scaling while running the
+benchmark.  Exactly how to do this depends on the Linux distribution,
+desktop environment, and installed programs.  Specific details are a moving
+target, so we will not attempt to exhaustively document them here.
+
+One simple option is to use the `cpupower` program to change the
+performance governor to "performance".  This tool is maintained along with
+the Linux kernel and provided by your distribution.
+
+It must be run as root, like this:
+
+```bash
+sudo cpupower frequency-set --governor performance
+```
+
+After this you can verify that all CPUs are using the performance governor
+by running this command:
 
 ```bash
-for f in /sys/devices/system/cpu/cpu?/cpufreq/scaling_governor
-  do
-  before=`cat $f`
-  echo "performance" > $f
-  ./mybench  # run your benchmark here
-  echo $before > $f
-done
+cpupower frequency-info -o proc
 ```
 
-note, you will have to run this as root (under sudo).
+The benchmarks you subsequently run will have less variance.
```

Comment:
we used to also suggest putting the governor back to how it was before, which was the intent of the script. so i think a fuller example (for completeness) that saves the current governor settings and resets them would be ideal.

though i understand that's not straightforward using cpupower's output. the best i have is

```bash
#!/bin/bash                                                                                                                                             
BENCHMARK=$1

CPUS=$(sudo cpupower frequency-info -o proc | tail -n +2 | tr -s ' ' | cut -d\  -f2,14)

declare -A prev_cpus=()

while IFS= read -r c
do
  IFS=' ' read CPU GOVERNOR <<< $c
  prev_cpus[$CPU]=$GOVERNOR
done <<< "$CPUS"

echo "setting all CPUs to 'performance'"
sudo cpupower frequency-set -g performance

echo "running $BENCHMARK"
$BENCHMARK

echo "setting all CPUs back to previous states"
for i in "${!prev_cpus[@]}"
do
  echo "setting cpu $i back to ${prev_cpus[$i]}"
  sudo cpupower --cpu $i frequency-set -g ${prev_cpus[$i]}
done
```

maybe it's worth having this checked in to `tools` and pointing folks at it.

Reply:
I did write a script like yours, but I found the workflow annoying. What I've always done in the past is set the machine to performance mode for the entire duration of an interactive benchmarking session. What I'd advocate for in the user guide is a description of how to achieve this workflow.

This isn't to say that a script supporting the "performance while running one command" must be excluded (see idea below).

In practice there is a delay setting the governors with `cpupower`, which makes sense because there is a short delay before they take full effect.  I've also got 24 cores, so cpupower spat out 48 lines of output each time.  The delay and the output made it annoying to use for every single benchmark invocation.  Using a script wrapper is also yet another thing to remember, integrate into a workflow (which may be script driven, etc.), forget to run, etc.

One idea: if the script had optional `--enable` and `--disable` options that saved and restored settings, respectively, both workflows could be supported.  Maybe it would save them somewhere under [`$XDG_STATE_HOME`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).  It could be used like this:

```
with_performance ./mybenchmark
```

or like this:

```
with_performance --enable
./mybenchmark
make
./mybenchmark
./make
./mybenchmark
etc.
with_performance --disable
```

so both workflows would be supported.

Reply:
makes total sense. perhaps having this conversation here will help people too :)